exercises/robot-name/robot-name-test.rkt

```diff
@@ -0,0 +1,38 @@
+#lang racket
+(require racket/class
+         "robot-name.rkt")
+
+(module+ test
+  (require rackunit
+           rackunit/text-ui)
+
+  (define proper-robot-name #px"\\p{Lu}{2}\\p{Nd}{4}")
+  
+  (define (check-proper-robot-name r)
+    (check-regexp-match proper-robot-name (send r get-name)))
+  
+  (define robot-name-suite 
+    (test-suite
+     "Check proper operation of 'robots'"
+     
+     (test-case
+      "Check that robot names conforms to the expected standard."
+      (check-proper-robot-name (new robot%)))
+
+     (test-case
+      "Check that resetting a robot name also gives a proper name."
+      (let ([robby (new robot%)])
+        (check-regexp-match proper-robot-name (send robby get-name))
+        (send robby reset)
+        (check-regexp-match proper-robot-name (send robby get-name))))
+
+     (test-case
+      "Check that robots are created with unique names."
+      (check-eq?
+       (set-count
+        (list->set
+         (for/list ([i (in-range 10)])
+           (send (new robot%) get-name))))
+       10))))
```

Comment:
I don't think checking 10 of them really guarantees that the names will be collision free.

We can always check all `(* 26 26 10 10 10)` of them :smiling_imp:

Reply:
This might require them providing a way to reset the current pool of used names.

Reply:
Running half of the possible cases `(* 26 26 10 10 5)` took about 16 seconds on my computer - but it is probably massive overkill. Checking 1000 runs in about 117 ms ... and is much more of a test than 10.

It is probably a good idea to provide a way to reset the name cache though ...